Automattic/jetpack
# VideoPress: make Pagination work with true data

Currently, the number of pages of the Pagination component is working with hardcoded data. This PR:
* Replaces the lib to request by native `fetch()`...
* To be able to get the response headers...
* to pick the `X-WP-Total` and `X-WP-TotalPages` fields...
* to be stored in the state.
* Finally, pass these data to the Pagination component to build the total of pages correctly.

Fixes https://github.com/Automattic/jetpack/issues/26098

#### Changes proposed in this Pull Request:

* VideoPress: make Pagination work with true data

#### Other information:

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Go to VideoPress dashboard
* Try to test with a site with several videos
* Confirm the pagination works as expected
* Test changing the page by searching, etc

<img width="822" alt="image" src="https://user-images.githubusercontent.com/77539/191579491-d01dae50-43c4-4b3a-bb3b-878ca33c3195.png">

<img width="449" alt="Screen Shot 2022-09-21 at 15 10 57" src="https://user-images.githubusercontent.com/77539/191579511-4529ffc6-b09c-43be-a442-e17e65e8ad03.png">
